SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialWorkerFlags/SpatialWorkerFlagsTest.cpp

```diff
@@ -51,39 +51,160 @@ SPATIALWORKERFLAGS_TEST(GIVEN_a_flagUpdate_op_WHEN_removing_a_worker_flag_THEN_f
+
+	TestTrue("Delegate Function was called only once", SpyObj->GetTimesFlagUpdated() == 1);
+
+	return true;
+}
+
+// any flag invoke
+
+SPATIALWORKERFLAGS_TEST(GIVEN_an_updated_flag_WHEN_registering_and_invoking_an_any_flag_update_delegate_THEN_delegate_is_invoked)
+{
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+
+	// Add test flag
+	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	// register callback
+	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
+	FOnAnyWorkerFlagUpdatedBP WorkerFlagDelegate;
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
+
+	SpatialWorkerFlags->RegisterAndInvokeAnyFlagUpdatedCallback(WorkerFlagDelegate);
+	// should invoke immediately
+	TestTrue("Delegate Function was called once", SpyObj->GetTimesFlagUpdated() == 1);
+
+	// Add test flag 2
+	Worker_FlagUpdateOp OpAddFlag2 = CreateWorkerFlagUpdateOp("test2", "20");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag2);
+
+	TestTrue("Delegate Function was called twice", SpyObj->GetTimesFlagUpdated() == 2);
+
+	return true;
+}
+
+SPATIALWORKERFLAGS_TEST(GIVEN_no_flags_WHEN_registering_and_invoking_an_any_flag_update_delegate_THEN_delegate_is_not_invoked)
+{
+	// register callback
+	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
+	FOnAnyWorkerFlagUpdatedBP WorkerFlagDelegate;
+	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetAnyFlagUpdated);
+
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+	SpatialWorkerFlags->RegisterAndInvokeAnyFlagUpdatedCallback(WorkerFlagDelegate);
+	// should not invoke
+	TestTrue("Delegate Function was not called", SpyObj->GetTimesFlagUpdated() == 0);
+
 	return true;
 }
 
-SPATIALWORKERFLAGS_TEST(GIVEN_a_bound_delegate_WHEN_unbind_the_delegate_THEN_bound_function_is_not_invoked)
+// a flag update
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_registered_flag_update_delegate_WHEN_the_worker_flag_updates_THEN_delegate_is_invoked)
 {
 	UWorkerFlagsTestSpyObject* SpyObj = NewObject<UWorkerFlagsTestSpyObject>();
-	FOnWorkerFlagsUpdatedBP WorkerFlagDelegate;
+	FOnWorkerFlagUpdatedBP WorkerFlagDelegate;
 	WorkerFlagDelegate.BindDynamic(SpyObj, &UWorkerFlagsTestSpyObject::SetFlagUpdated);
 
 	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
-	SpatialWorkerFlags->BindToOnWorkerFlagsUpdated(WorkerFlagDelegate);
+	const FString TestFlag = TEXT("test");
+	SpatialWorkerFlags->RegisterFlagUpdatedCallback(TestFlag, WorkerFlagDelegate);
+
 	// Add test flag
-	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp("test", "10");
+	Worker_FlagUpdateOp OpAddFlag = CreateWorkerFlagUpdateOp(TCHAR_TO_ANSI(*TestFlag), "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
+
+	// Update test flag
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(OpAddFlag);
```

Comment:
Huh, odd this fires again even if the value has changed. Didn't realize. Interesting

Reply:
This is the behaviour that was in place not sure if desirable..

Reply:
Oh yeah, not saying to change it just something I found funny.